rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -55,9 +57,30 @@ enum class publishing_mode_t
   AUTO           // Use publishing mode set in XML file or Fast DDS default
 };
 
+class CustomTopicListener final : public eprosima::fastdds::dds::TopicListener
+{
+public:
+  RMW_FASTRTPS_SHARED_CPP_PUBLIC
+  explicit CustomTopicListener(EventListenerInterface * event_listener);
+
+  RMW_FASTRTPS_SHARED_CPP_PUBLIC
+  void
+  on_inconsistent_topic(
+    eprosima::fastdds::dds::Topic * topic,
+    eprosima::fastdds::dds::InconsistentTopicStatus status) override;
+
+  void add_event_listener(EventListenerInterface * event_listener);
+
+  void remove_event_listener(EventListenerInterface * event_listener);
+
+private:
+  std::set<EventListenerInterface *> event_listeners_;
```

Comment:
Should this collection be protected against access from different threads?

Reply:
Yes, good idea.  Fixed in 222d3ce43fffe6191a7ee544bb21bce915f688e1.